tests/utils/test_run_dbt_cloud_job.py

```diff
@@ -0,0 +1,131 @@
+import json
+from pathlib import Path
+
+import pytest
+from responses import Response
+
+from utils.run_dbt_cloud_job import trigger_dbt_run, poll_dbt_run
+
+
+def test_should_trigger_run(responses):
+    # GIVEN: DBT cloud ready to accept job
+    rsp = Response(method="POST", url="https://cloud.getdbt.com/api/v2/accounts/1001/jobs/101/run/",
+                   status=200, headers={"Authorization": f"Token test-dbt-key", "Content-Type": "application/json"},
+                   body=with_response_body("job.200.json"))
+    responses.add(rsp)
+
+    # WHEN: trigger job
+    run_id = trigger_dbt_run(101, "Test Job")
+
+    # THEN: expect request to have been sent
+    responses.assert_call_count("https://cloud.getdbt.com/api/v2/accounts/1001/jobs/101/run/", 1)
+    assert json.loads(responses.calls[0].request.body) == {"cause": "Airflow run - Test Job"}
+
+    # THEN: expect run id to have been properly loaded from response
+    assert run_id == 10000
+
+
+@pytest.mark.parametrize("status,content", [(400, 'job.400.json'), (404, 'job.404.json')])
+def test_should_handle_failure(responses, status, content):
+    # GIVEN: DBT cloud returns a non-200 response
+    rsp = Response(method="POST", url="https://cloud.getdbt.com/api/v2/accounts/1001/jobs/101/run/",
+                   status=status, headers={"Authorization": f"Token test-dbt-key", "Content-Type": "application/json"},
+                   body=with_response_body(content))
+    responses.add(rsp)
```

Comment:
@ifoukarakis , can we factor out this block into a method and pass the body content as a param?

Reply:
That's a really good idea. I used the [factories as fixtures](https://docs.pytest.org/en/7.1.x/how-to/fixtures.html?highlight=factories#factories-as-fixtures) pattern.